src/coreclr/jit/codegencommon.cpp

```diff
@@ -4191,6 +4198,34 @@ void CodeGen::genEnregisterIncomingStackArgs()
                 }
             }
         }
+#elif defined(TARGET_RISCV64)
+        {
+            bool FPbased;
+            int  base = compiler->lvaFrameAddress(varNum, &FPbased);
+
+            if (emitter::isValidSimm12(base))
+            {
+                GetEmitter()->emitIns_R_S(ins_Load(regType), emitTypeSize(regType), regNum, varNum, 0);
+            }
+            else
+            {
+                if (tmp_reg == REG_NA)
+                {
+                    regNumber reg2 = FPbased ? REG_FPBASE : REG_SPBASE;
+                    tmp_offset     = base;
+                    tmp_reg        = REG_RA; // TODO CHECK R21 => RA
+
+                    GetEmitter()->emitIns_I_la(EA_PTRSIZE, REG_RA, base);                   // TODO CHECK R21 => RA
+                    GetEmitter()->emitIns_R_R_R(INS_add, EA_PTRSIZE, REG_RA, REG_RA, reg2); // TODO CHECK R21 => RA
+                    GetEmitter()->emitIns_R_S(ins_Load(regType), emitTypeSize(regType), regNum, varNum, -8);
+                }
+                else
+                {
+                    int baseOffset = -(base - tmp_offset) - 8;
+                    GetEmitter()->emitIns_R_S(ins_Load(regType), emitTypeSize(regType), regNum, varNum, baseOffset);
+                }
+            }
```

Comment:
Why is this special casing on RISCV (and LA64, for that matter) necessary? It looks strange to me -- if RISCV cannot handle large displacements, then how does this handle the situation given that `tmp_offset`/`tmp_reg` is only updated once?

Other platforms use `CodeGen::rsGetRsvdReg` to handle reserving a temporary register to be used in situations of unencodable offsets. Can RISCV/LA64 be put on the same plan?

Reply:
Thank you. Removed codes and replaced temp reg use with rsGetRsvdReg.